### PR TITLE
Fe fix/toggle save

### LINKED ISSUE #21: Component - ToggleSave

### PR DESCRIPTION
**Pull request by andreasorensen**
# Description

- Fixed ToggleSave component to leverage the cache of savedUserMushrooms
-- ToggleSave now utilized the savedUserCache to determine whether a mushroom is already saved & if it is, the user is shown the Saved icon, if not, it will show the notSaved icon.
-- It also checks to see if the mushroom is in the savedUserMushrooms & if it is, when the toggle is clicked, it removes it. If it is not, it adds it. 
-- If you're on the FieldGuidePage, the page is immediately rerendered with the usersSavedMushrooms.

Closes #62 
Closes #46 
Closes #21 

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update
- [ ] Testing
- [X] Refactoring

# How Has This Been Tested?

I'm getting to Cypress next!

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Elise-Jones — APPROVED** [review]
You just found and unlocked the power of graphql allowing us to hit the api only when actuallly needed. Such great work on this issue and researching.

File: src/Components/ToggleSave/ToggleSave.js [hunk 1]
```diff
@@ -1,35 +1,60 @@
...
-  const userId = 3;
-
-  const handleToggle = async () => {
-    try {
-      console.log("User ID:", userId); 
-      console.log("Mushroom ID:", mushroomId);
-      
-      await toggleSaveMushroom({
-        variables: { userId, mushroomId },
+import { useMutation, useQuery } from "@apollo/client";
```
Comment (jmartin777) [inline src/Components/ToggleSave/ToggleSave.js:6]: now using both mutation and query functionalities from Apollo Client.!

really cool

File: src/Components/ToggleSave/ToggleSave.js [hunk 2]
```diff
@@ -1,35 +1,60 @@
...
+import { SAVE_DELETE_USER_MUSHROOM, GET_SAVED_USERMUSHROOMS } from "../../queries";
```
Comment (jmartin777) [inline src/Components/ToggleSave/ToggleSave.js:7]: Updating  Cache!: Really cool how you update the Apollo Client cache. Depending on whether the mushroom was saved or deleted, this updates the list of saved mushrooms for the user in the cache. 

cache.readQuery and cache.writeQuery is really powerful with graphQL

**Review by jmartin777 — APPROVED** [review]
Great Refactor here - beginning with a fresh slate and coming back into this App. with Apollo / GraphQl. Huge light bulb moment here!

**Pull request merged**